lib/index.js

```diff
@@ -507,14 +527,13 @@ Socket.prototype._flush = function() {
     };
 
     if (flags & zmq.ZMQ_POLLIN) {
-      emitArgs = ['message'];
+      emitArgs = this._read();
 
```

Comment:
do you mean `this.read()`? i'm not sure we should be calling `_read()` directly.